src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -1555,6 +1555,166 @@ class DropMaterializedViewStatementSegment(BaseSegment):
+    """
+
+    type = "create_database_statement"
+
+    match_grammar = StartsWith(Sequence("CREATE", "DATABASE"))
+
+    parse_grammar = Sequence(
+        "CREATE",
+        "DATABASE",
+        Ref("DatabaseReferenceSegment"),
+        Ref.keyword("WITH", optional=True),
+        AnyNumberOf(
+            Sequence(
+                "OWNER",
+                Ref("EqualsSegment", optional=True),
+                Ref("ObjectReferenceSegment"),
+            ),
+            Sequence(
+                "TEMPLATE",
+                Ref("EqualsSegment", optional=True),
+                Ref("ObjectReferenceSegment"),
+            ),
+            Sequence(
+                "ENCODING",
+                Ref("EqualsSegment", optional=True),
+                OneOf(Ref("QuotedLiteralSegment"), "DEFAULT"),
+            ),
+            OneOf(
+                # LOCALE This is a shortcut for setting LC_COLLATE and LC_CTYPE at once.
+                # If you specify this, you cannot specify either of those parameters.
+                Sequence(
+                    "LOCALE",
+                    Ref("EqualsSegment", optional=True),
+                    Ref("QuotedLiteralSegment"),
+                ),
+                AnyNumberOf(
+                    Sequence(
+                        "LC_COLLATE",
+                        Ref("EqualsSegment", optional=True),
+                        Ref("QuotedLiteralSegment"),
+                    ),
+                    Sequence(
+                        "LC_CTYPE",
+                        Ref("EqualsSegment", optional=True),
+                        Ref("QuotedLiteralSegment"),
+                    ),
+                ),
+            ),
+            Sequence(
+                "TABLESPACE",
+                Ref("EqualsSegment", optional=True),
+                Ref("ParameterNameSegment"),
+            ),
+            Sequence(
+                "ALLOW_CONNECTIONS",
+                Ref("EqualsSegment", optional=True),
+                Ref("BooleanLiteralGrammar"),
+            ),
+            Sequence(
+                "CONNECTION",
+                "LIMIT",
+                Ref("EqualsSegment", optional=True),
+                Ref("NumericLiteralSegment"),
+            ),
+            Sequence(
+                "IS_TEMPLATE",
+                Ref("EqualsSegment", optional=True),
+                Ref("BooleanLiteralGrammar"),
+            ),
+        ),
+    )
+
+
+@postgres_dialect.segment()
+class AlterDatabaseStatementSegment(BaseSegment):
+    """A `ALTER DATABASE` statement.
+
+    As specified in https://www.postgresql.org/docs/14/sql-alterdatabase.html
+    """
+
+    type = "alter_database_statement"
+
+    match_grammar = StartsWith(Sequence("ALTER", "DATABASE"))
+
+    parse_grammar = Sequence(
+        "ALTER",
+        "DATABASE",
+        Ref("DatabaseReferenceSegment"),
+        OneOf(
+            Sequence(
+                Sequence("WITH", optional=True),
```

Comment:
```suggestion
                Ref.keyword("WITH", optional=True),
```
To be consistent with CREATE/DROP